### PR TITLE
add support for the _curses, bz2, and grp modules (mercurial needs them)

### PR DESCRIPTION
**Pull request by toshok**

### CODE REVIEW

**Comment by toshok** [comment]
```
                           649e33f464679d2a02:  96ad40344213ae02da:
       django_template.py             3.9s (4)             4.1s (6)  +6.5%
            pyxl_bench.py             3.8s (4)             3.8s (6)  -1.5%
sqlalchemy_imperative2.py             4.5s (4)             4.5s (6)  +0.4%
        django_migrate.py             1.7s (4)             1.7s (6)  +0.0%
      virtualenv_bench.py             4.9s (4)             5.0s (6)  +0.5%
                  geomean                 3.5s                 3.6s  +1.1%
```

the 6% increase in django_template.py seems to be the grp module's presence.

**Comment by toshok** [comment]
okay, reworked the check in GCVisitor::visit a bit (and switched the non-heap root test to an assert).

also added some gc methods to `section_ordering.txt` since they're all small and hot and call one another.

```
                           649e33f464679d2a02:  9759cd7cf453b81a82:
       django_template.py             3.9s (4)             3.8s (2)  -0.8%
            pyxl_bench.py             3.8s (4)             3.8s (2)  -0.9%
sqlalchemy_imperative2.py             4.5s (4)             4.5s (2)  -0.6%
        django_migrate.py             1.7s (4)             1.7s (2)  +0.6%
      virtualenv_bench.py             4.9s (4)             4.8s (2)  -2.0%
                  geomean                 3.5s                 3.5s  -0.7%
```

**Comment by kmod** [comment]
lgtm

File: from_cpython/CMakeLists.txt [hunk 1]
```diff
@@ -117,6 +117,10 @@ add_custom_command(OUTPUT
                        ${CMAKE_BINARY_DIR}/lib_pyston/_multiprocessing.pyston.so
                        ${CMAKE_BINARY_DIR}/lib_pyston/pyexpat.pyston.so
                        ${CMAKE_BINARY_DIR}/lib_pyston/_elementtree.pyston.so
+                       ${CMAKE_BINARY_DIR}/lib_pyston/bz2.so
+                       ${CMAKE_BINARY_DIR}/lib_pyston/grp.so
+                       ${CMAKE_BINARY_DIR}/lib_pyston/termios.so
+                       ${CMAKE_BINARY_DIR}/lib_pyston/_curses.so
```
Comment (kmod) [inline from_cpython/CMakeLists.txt]: I'm seeing cmake rebuild these each time -- I think these should be .pyston.so?

Comment (toshok) [inline from_cpython/CMakeLists.txt]: ah, quite right.  fix incoming

**Pull request merged**